Document maximum number of IBFT 2.0 validators (#895)
hyperledger/besu-docs: docs/HowTo/Configure/Consensus-Protocols/IBFT.md

[add or remove validators](Add-Validators.md#ibft-20). Adding or removing a validator
requires a majority vote (greater than 50%) of validators.

## Validators

### Minimum number of validators

IBFT 2.0 requires four validators to be Byzantine fault tolerant. Byzantine fault tolerance is the
ability for a blockchain network to function correctly and reach consensus despite nodes failing or
propagating incorrect information to peers.

### Maximum number of validators

As the number of validators increase, the message complexity increases as well, which can cause a loss of performance.
In [network tests](https://wiki.hyperledger.org/display/BESU/Maximum+Validator+count+for+an+IBFT2+Network), IBFT 2.0 handles up to 30 validators safely with no loss of performance.

> 

Reviewer: Is it worth clarifying here that the number of non-validator nodes doesn’t affect the performance?

Reviewer: I'm concerned that someone might read this to mean you can only have max 30 nodes in a besu network

[add or remove validators](Add-Validators.md#ibft-20). Adding or removing a validator
requires a majority vote (greater than 50%) of validators.

## Validators

### Minimum number of validators

IBFT 2.0 requires four validators to be Byzantine fault tolerant. Byzantine fault tolerance is the
ability for a blockchain network to function correctly and reach consensus despite nodes failing or
propagating incorrect information to peers.

### Maximum number of validators

As the number of validators increase, the message complexity increases as well, which can cause a loss of performance.
In [network tests](https://wiki.hyperledger.org/display/BESU/Maximum+Validator+count+for+an+IBFT2+Network), IBFT 2.0 handles up to 30 validators safely with no loss of performance.

> After 35 validators, transaction pool time grows exponentially.

Reviewer 2: Think we should just remove this last sentence "After 35 validators, transaction pool time grows exponentially.". We haven't finished analysis on why it's failing at this point.

Author: @jframe All set, could you re-review? Do you know if these limits also apply to QBFT?

Reviewer 2: The first part "As the number of validators increase, the message complexity increases as well, which can cause a loss of performance." does apply to QBFT as well. But since we haven't done this network test for QBFT we can't say at what point QBFT stops functioning well though it's quite likely to be the same number.

> As the number of validators increase, the message complexity increases as well, which can cause a loss of performance.
> In [network tests](https://wiki.hyperledger.org/display/BESU/Maximum+Validator+count+for+an+IBFT2+Network), IBFT 2.0 handles up to 30 validators safely with no loss of performance.
> Non-validator nodes do not affect performance and do not count towards the maximum limit.
> 

Reviewer 3: ```suggestion
As the number of validators increase, the message complexity increases, which can decrease performance.
In [network tests](https://wiki.hyperledger.org/display/BESU/Maximum+Validator+count+for+an+IBFT2+Network), IBFT 2.0 handles up to 30 validators with no loss of performance.

Non-validator nodes don't affect performance and don't count towards the maximum limit.

```
